File: day1/solution.js
"use strict";
const { readFileSync } = require("fs");
const file = readFileSync("./input.txt", "utf-8");
const fileArray = file.split(/\n/);
const caloriesPerElf = [];
let caloriesCurrentElf = 0;
fileArray.forEach((line) => {
    if (line != "") {
        caloriesCurrentElf += parseInt(line);
    }
    else {
        caloriesPerElf.push(caloriesCurrentElf);
        caloriesCurrentElf = 0;
    }
});
console.log(Math.max(...caloriesPerElf));
const sortedCaloriesPerElf = caloriesPerElf
    .sort((a, b) => a - b)
    .reverse();
console.log(sortedCaloriesPerElf.slice(0, 3).reduce((a, b) => a + b));

File: day1/solution.py
with open("input.txt") as f:
    total_calories_per_elf = []
    total_calories_current_elf = 0
    for line in f:
        if line != "\n":
            total_calories_current_elf += int(line)
        else:
            total_calories_per_elf.append(total_calories_current_elf)
            total_calories_current_elf = 0

print(max(total_calories_per_elf))
top_three_elves = sorted(total_calories_per_elf, reverse=True)[:3]
print(sum(top_three_elves))
